### PR TITLE
Adding to_csv functions to the query library

### PR DESCRIPTION
**Pull request by AspectVI**
### Submitter:

Have you (where applicable):

* [ ] Added unit tests?
* [ ] Checked the latest commit runs on Dev?
* [ ] Updated the example config file(s) and README?

---

### Reviewer

Does this PR:

* [ ] Place non-StackStorm code into the `lib` directory?
* [ ] Have unit tests for the action/sensor and `lib` layers?
* [ ] Have clear and obvious action parameter names and descriptions?

### CODE REVIEW

**Review by anish-mudaraddi — APPROVED** [review]

**Review by khalford — CHANGES_REQUESTED** [review]

File: tests/lib/openstack_query/query_blocks/test_query_output.py [hunk 1]
```diff
@@ -7,6 +8,58 @@
...
+                "server_id": "server_id1",
+                "server_name": "server1",
+                "user_id": "user_id1",
+                "user_name": "user1",
+            },
+            {
+                "server_id": "server_id2",
+                "server_name": "server2",
+                "user_id": "user_id1",
+                "user_name": "user1",
```
[the remark this answers is not in the document]
Comment (AspectVI) [reply tests/lib/openstack_query/query_blocks/test_query_output.py:25]: changed

File: tests/lib/openstack_query/query_blocks/test_query_output.py [hunk 2]
```diff
@@ -7,6 +8,58 @@
...
+                "server_id": "server_id3",
+                "server_name": "server3",
+                "user_id": "user_id2",
+                "user_name": "user2",
+            },
+            {
+                "server_id": "server_id4",
+                "server_name": "server4",
+                "user_id": "user_id2",
+                "user_name": "user2",
```
[the remark this answers is not in the document]
Comment (AspectVI) [reply tests/lib/openstack_query/query_blocks/test_query_output.py:39]: changed

File: tests/lib/openstack_query/query_blocks/test_query_output.py [hunk 3]
```diff
@@ -7,6 +8,58 @@
...
+                "user_id": "user_id1",
+                "user_name": "user1",
+            },
+        ],
+        "user_name is user2": [
+            {
+                "server_id": "server_id3",
+                "server_name": "server3",
+                "user_id": "user_id2",
+                "user_name": "user2",
```
[the remark this answers is not in the document]
Comment (AspectVI) [reply tests/lib/openstack_query/query_blocks/test_query_output.py:33]: altered

File: lib/openstack_query/query_blocks/query_output.py [hunk 4]
```diff
@@ -242,3 +244,55 @@ def _flatten_list(data: List[Dict]) -> Dict:
...
+
+    def to_csv_dictionary(self, data: Union[List, Dict], dir_path):
+        """
+        Takes a dictionary of dictionaries and outputs them into separate designated csv file 'self._parse_properties'
+        :param data: this is the dictionary of dictionaries passed in to this function
+        :param dir_path: this is the output path the csv files will be created in
+        :return: Does it return anything.
+        """
+
+        for p_id, p_info in data.items():
```
[the remark this answers is not in the document]
Comment (AspectVI) [reply lib/openstack_query/query_blocks/query_output.py:276]: After a further look it seem to be the meta data of the dict/lists that are past in. Then this is used to loop the correct amount of times and then later assign the names as the file name for the created files.

[the remark this answers is not in the document]
Comment (anish-mudaraddi) [reply lib/openstack_query/query_blocks/query_output.py:276]: yea we're looping through group name and the group items - which is a list of query results that belong the that group. 
so maybe `group_name`, `group_items`

[the remark this answers is not in the document]
Comment (AspectVI) [reply lib/openstack_query/query_blocks/query_output.py:276]: Cool, ill do that

**Review by khalford — APPROVED** [review]

**Comment by khalford** [comment]
LGTM

**Comment by AspectVI** [comment]
@khalford @anish-mudaraddi Am I fin to merge this PR, the codecov is falling over itself and causing test fails.

**Pull request merged**